src/WopiValidator.Core/Validators/ResponseContentValidator.cs

```diff
@@ -36,7 +36,18 @@ public ValidationResult Validate(IResponseData data, IResourceManager resourceMa
 			MemoryStream expectedContent;
 			if (!string.IsNullOrEmpty(ExpectedBodyContent))
 			{
-				expectedContent = new MemoryStream(Encoding.UTF8.GetBytes(ExpectedBodyContent));
+				string setting = null;
+				if (ExpectedBodyContent.StartsWith(Constants.StateOverrides.StateToken))
+					setting = ExpectedBodyContent.Substring(Constants.StateOverrides.StateToken.Length);
+
+				string expectedBodyContent = ExpectedBodyContent;
+				if (!String.IsNullOrEmpty(setting) &&
+					!savedState.TryGetValue(setting, out expectedBodyContent))
+				{
+					throw new InvalidOperationException("OverrideUrl specified in definition but not found in state dictionary.  Did it depend on a request that failed?");
+				}
+
+				expectedContent = new MemoryStream(Encoding.UTF8.GetBytes(expectedBodyContent));
```

Comment:
What is this doing?

Reply:
In test cases, I added logic to verify file content is updated by checking GetFile response content is equal to PutFile request content. There's a state to store PutFile request content, expected content of GetFile is the saved state.

Reply:
This seems unnecessary, though. Test cases can used fixed content that can be verified statically rather than dynamically.